src/rez_pip/pip.py

```diff
@@ -57,6 +59,36 @@ def name(self) -> str:
     def version(self) -> str:
         return self.metadata.version
 
+    def isLocal(self) -> bool:
+        """Returns True if the wheel is a local file"""
+        return urllib.parse.urlparse(self.download_info.url).scheme == "file"
+
+    @property
+    def path(self) -> str:
+        if not self.isLocal():
+            raise RuntimeError("Cannot be path of a non local wheel")
+
+        # Taken from https://github.com/python/cpython/pull/107640
+        path = self.download_info.url[5:]
+        if path[:3] == "///":
+            # Remove empty authority
+            path = path[2:]
+        elif path[:12] == "//localhost/":
+            # Remove 'localhost' authority
+            path = path[11:]
+        if path[:3] == "///" or (path[:1] == "/" and path[2:3] in ":|"):
+            # Remove slash before DOS device/UNC path
+            path = path[1:]
+        if path[1:2] == "|":
+            # Replace bar with colon in DOS drive
+            path = path[:1] + ":" + path[2:]
+
+        _path = pathlib.PurePath(os.fsdecode(urllib.parse.unquote_to_bytes(path)))
+        if not _path.is_absolute():
+            raise ValueError(f"URI is not absolute: {self.download_info.url!r}")
+
+        return os.fspath(path)
```

Comment:
The return value is bad when there is a space on Windows. See https://github.com/JeanChristopheMorinPerso/rez-pip/issues/74#issuecomment-1823146275.